### ISSUE TITLE
Build error with DoublePrec_CPU=OFF

### ISSUE THREAD

**Issue by FilipeMaia** [3dem/relion#919]
When setting DoublePrec_CPU=OFF in cmake I get the following error:

```c++
/davinci/Cellar/RELION/relion/src/ml_optimiser.cpp:6500:45: error: no match for ‘operator=’ (operand types are ‘__gnu_cxx::__alloc_traits<std::allocator<MultidimArray<float> >, MultidimArray<float> >::value_type’ {aka ‘MultidimArray<float>’} and ‘MultidimArray<double>’)
                 exp_local_STMulti[img_id] = STmult;
                                             ^~~~~~
In file included from /davinci/Cellar/RELION/relion/src/fftw.h:50,
                 from /davinci/Cellar/RELION/relion/src/projector.h:23,
                 from /davinci/Cellar/RELION/relion/src/ml_model.h:23,
                 from /davinci/Cellar/RELION/relion/src/ml_optimiser.h:33,
                 from /davinci/Cellar/RELION/relion/src/ml_optimiser.cpp:45:
/davinci/Cellar/RELION/relion/src/multidim_array.h:4420:23: note: candidate: ‘MultidimArray<T>& MultidimArray<T>::operator=(const MultidimArray<T>&) [with T = float]’
     MultidimArray<T>& operator=(const MultidimArray<T>& op1)
                       ^~~~~~~~
/davinci/Cellar/RELION/relion/src/multidim_array.h:4420:23: note:   no known conversion for argument 1 from ‘MultidimArray<double>’ to ‘const MultidimArray<float>&’
```

This suggests that line: https://github.com/3dem/relion/blob/04079953afd9f74c8705e24004091575b4fd71e0/src/ml_optimiser.cpp#L6494
should be replaced with:
```c++
MultidimArray<RFLOAT> STmult;
```

**Environment:**
 - OS: CentOS Linux release 7.9.2009 (Core)
 - MPI runtime: OpenMPI 3.1.3
 - RELION version 4.0.0 (04079953afd9f74c8705e24004091575b4fd71e0)

**Comment by FilipeMaia** [comment]
I just confirmed it builds correctly with the proposed change.

**Comment by biochem-fan** [comment]
Thank you very much for your report.
I committed your proposed fix 2daa7447, which is in the `ver4.0` branch.
